ggml/src/ggml-cpu.c

```diff
@@ -11641,79 +11641,189 @@ static void ggml_compute_forward_add_rel_pos(
     }
 }
 
-// ggml_compute_forward_rwkv_wkv
+// ggml_compute_forward_rwkv_wkv6
 
-static void ggml_compute_forward_rwkv_wkv_f32(
+static void ggml_compute_forward_rwkv_wkv6_f32(
         const struct ggml_compute_params * params,
         struct ggml_tensor * dst) {
-    const size_t T = dst->src[1]->ne[3];
-    const size_t C = dst->ne[0];
-    const size_t H = dst->src[1]->ne[2];
-    const size_t n_seqs = dst->src[5]->ne[1];
+    const int64_t T = dst->src[1]->ne[3];
+    const int64_t C = dst->ne[0];
+    const int64_t HEADS = dst->src[1]->ne[2];
+    const int64_t n_seqs = dst->src[5]->ne[1];
+    const int64_t head_size = C / HEADS;
 
     float * dst_data = (float *) dst->data;
     float * state = ((float *) dst->data) + C * T;
 
-    if (params->ith != 0) {
+    if ((int64_t)params->ith >= HEADS) {
         return;
     }
 
-    memset(dst_data, 0, T * C * sizeof(float));
+    int64_t h_start = (HEADS * params->ith) / params->nth;
+    int64_t h_end = ((HEADS * (params->ith + 1)) / params->nth < HEADS) ? 
+                (HEADS * (params->ith + 1)) / params->nth : HEADS;
 
```

Comment:
For consistency, rewrite this to be more inline with the common pattern for distributing threads. For example, something similar in style to:

https://github.com/ggerganov/llama.cpp/blob/ce027adfb3b131f0d2368294fc276bb0e342b3f6/ggml/src/ggml-cpu.c#L4037-L4054